app/scripts/lib/ppom/ppom-util.ts

```diff
@@ -37,18 +39,37 @@ type PPOMRequest = Omit<JsonRpcRequest, 'method' | 'params'> & {
   method: typeof METHOD_SEND_TRANSACTION;
   params: [TransactionParams];
 };
+
 export async function validateRequestWithPPOM({
   ppomController,
   request,
   securityAlertId,
   chainId,
+  updateSecurityAlertResponse: updateSecurityResponse,
 }: {
   ppomController: PPOMController;
   request: JsonRpcRequest;
   securityAlertId: string;
   chainId: Hex;
+  updateSecurityAlertResponse: (
+    method: string,
+    securityAlertId: string,
+    securityAlertResponse: SecurityAlertResponse,
+  ) => void;
 }): Promise<SecurityAlertResponse> {
   try {
+    if (!(await isChainSupported(chainId))) {
+      return {
+        ...SECURITY_ALERT_RESPONSE_CHAIN_NOT_SUPPORTED,
+        securityAlertId,
+      };
+    }
+
+    updateSecurityResponse(request.method, securityAlertId, {
```

Comment:
Isn't this asynchronous, should we await this before continuing?

Reply:
You are right, the type previous was void but because we were intentionally not awaiting. Fixed.